src/Microsoft.VisualStudio.Threading.Analyzers/MultilingualResources/Microsoft.VisualStudio.Threading.Analyzers.it.xlf

```diff
@@ -27,14 +27,17 @@
           <target state="translated">Evita attese sincrone problematiche</target>
         </trans-unit>
         <trans-unit id="VSTHRD010_MessageFormat" translate="yes" xml:space="preserve">
-          <source>Visual Studio service "{0}" should be used on main thread explicitly.
-Call ThreadHelper.ThrowIfNotOnUIThread() or await JoinableTaskFactory.SwitchToMainThreadAsync() first.</source>
-          <target state="translated">Il servizio "{0}" di Visual Studio non deve essere usato sul thread principale in modo esplicito.
+          <source>The type "{0}" should be used on main thread explicitly.
+Call {1}() or await JoinableTaskFactory.SwitchToMainThreadAsync() first.</source>
```

Comment:
nit: should `JoinableTaskFactory.SwitchToMainThreadAsync` also be generalized here, since users can provide main thread switching methods via data as well?

Reply:
What do you mean by "via data"? Per the threading rules, the only recommended way to switch to the UI thread is via this method. So it seemed OK to put it in the doc directly. But I'm open to other ideas.

Reply:
Sorry, I should have been a bit clearer - you define some main switching functions in the `vs-threading.MainThreadSwitchingMethods.txt` file. But if it's much preferred to use JTF.SwitchToMainThreadAsync then maybe it should stay.

Reply:
Oh, that. Well, the other method in that file is only there for legacy reasons and is cps specific. Really this is the only one I want to encourage.